File: web/src/types/ModelType.ts
/// <summary>
/// UndertaleModTool refers to asset types as 'models' in
/// the <c>UndertaleModLib.Models</c> namespace, see also
/// the <see cref="UndertaleModLib.UndertaleData" /> properties
/// </summary>
export const ModelType = {
	/// <summary>
	/// General info of the data file.
	/// </summary>
	GeneralInfo: 0,

	/// <summary>
	/// General Options of the data file.
	/// </summary>
	Options: 1,

	/// <summary>
	/// Languages of the data file.
	/// </summary>
	Language: 2,

	/// <summary>
	/// The used extensions of the data file.
	/// </summary>
	Extensions: 3,

	/// <summary>
	/// The used sounds of the data file.
	/// </summary>
	Sounds: 4,

	/// <summary>
	/// The audio groups of the data file.
	/// </summary>
	AudioGroups: 5,

	/// <summary>
	/// The sprites of the data file.
	/// </summary>
	Sprites: 6,

	/// <summary>
	/// The backgrounds (or Tilesets) of the data file.
	/// </summary>
	Backgrounds: 7,

	/// <summary>
	/// The paths of the data file.
	/// </summary>
	Paths: 8,

	/// <summary>
	/// The scripts of the data file.
	/// </summary>
	Scripts: 9,

	/// <summary>
	/// The global initialization scripts of the data file.
	/// </summary>
	GlobalInitScripts: 10,

	/// <summary>
	/// The global end scripts of the data file.
	/// </summary>
	GameEndScripts: 12,

	/// <summary>
	/// The used shaders of the data file.
	/// </summary>
	Shaders: 13,

	/// <summary>
	/// The fonts of the data file.
	/// </summary>
	Fonts: 14,

	/// <summary>
	/// The Timelines of the data file.
	/// </summary>
	Timelines: 15,

	/// <summary>
	/// The game objects of the data file.
	/// </summary>
	GameObjects: 16,

	/// <summary>
	/// The rooms of the data file.
	/// </summary>
	Rooms: 17,

	/// <summary>
	/// The texture page items from the data file.
	/// </summary>
	TexturePageItems: 18,

	/// <summary>
	/// The code entries of the data file.
	/// </summary>
	Code: 19,

	/// <summary>
	/// The used variables of the data file.
	/// </summary>
	Variables: 20,

	/// <summary>
	/// The functions of the data file.
	/// </summary>
	Functions: 21,

	/// <summary>
	/// The code locals of the data file.
	/// </summary>
	CodeLocals: 22,

	/// <summary>
	/// The used strings of the data file.
	/// </summary>
	Strings: 23,

	/// <summary>
	/// The embedded images of the data file. This is used to store built-in particle sprites,
	/// every time you use <c>part_sprite</c> functions.
	/// </summary>
	EmbeddedImages: 24,

	/// <summary>
	/// The embedded textures of the data file.
	/// </summary>
	EmbeddedTextures: 25,

	/// <summary>
	/// The texture group infos of the data file.
	/// </summary>
	TextureGroupInfo: 26,

	/// <summary>
	/// The embedded audio of the data file.
	/// </summary>
	EmbeddedAudio: 27,

	/// <summary>
	/// The used tags of the data file.
	/// </summary>
	Tags: 28,

	/// <summary>
	/// The animation curves of the data file.
	/// </summary>
	AnimationCurves: 29,

	/// <summary>
	/// The sequences of the data file.
	/// </summary>
	Sequences: 30,

	/// <summary>
	/// The feature flags stored in the data file.
	/// </summary>
	FeatureFlags: 31,

	/// <summary>
	/// The filter effects stored in the data file.
	/// </summary>
	FilterEffects: 32,

	/// <summary>
	/// The particle systems stored in the data file.
	/// </summary>
	ParticleSystems: 33,

	/// <summary>
	/// The particle system emitters stored in the data file.
	/// </summary>
	ParticleSystemEmitters: 34,
} as const;

export type ModelTypeKeys = (typeof ModelType)[keyof typeof ModelType];

File: src/ModelType.cs
using UndertaleModLib;

namespace UndertaleModToolWASM;

/// <summary>
/// UndertaleModTool refers to asset types as 'models' in
/// the <c>UndertaleModLib.Models</c> namespace, see also
/// the <see cref="UndertaleData" /> properties
/// </summary>
public enum ModelType
{
    /// <seealso cref="UndertaleData.GeneralInfo"/>
    GeneralInfo = 0,

    /// <seealso cref="UndertaleData.Options"/>
    Options = 1,

    /// <seealso cref="UndertaleData.Language"/>
    Language = 2,

    /// <seealso cref="UndertaleData.Extensions"/>
    Extensions = 3,

    /// <seealso cref="UndertaleData.Sounds"/>
    Sounds = 4,

    /// <seealso cref="UndertaleData.AudioGroups"/>
    AudioGroups = 5,

    /// <seealso cref="UndertaleData.Sprites"/>
    Sprites = 6,

    /// <seealso cref="UndertaleData.Backgrounds"/>
    Backgrounds = 7,

    /// <seealso cref="UndertaleData.Paths"/>
    Paths = 8,

    /// <seealso cref="UndertaleData.Scripts"/>
    Scripts = 9,

    /// <seealso cref="UndertaleData.GlobalInitScripts"/>
    GlobalInitScripts = 10,

    /// <seealso cref="UndertaleData.GameEndScripts"/>
    GameEndScripts = 12,

    /// <seealso cref="UndertaleData.Shaders"/>
    Shaders = 13,

    /// <seealso cref="UndertaleData.Fonts"/>
    Fonts = 14,

    /// <seealso cref="UndertaleData.Timelines"/>
    Timelines = 15,

    /// <seealso cref="UndertaleData.GameObjects"/>
    GameObjects = 16,

    /// <seealso cref="UndertaleData.Rooms"/>
    Rooms = 17,

    /// <seealso cref="UndertaleData.TexturePageItems"/>
    TexturePageItems = 18,

    /// <seealso cref="UndertaleData.Code"/>
    Code = 19,

    /// <seealso cref="UndertaleData.Variables"/>
    Variables = 20,

    /// <seealso cref="UndertaleData.Functions"/>
    Functions = 21,

    /// <seealso cref="UndertaleData.CodeLocals"/>
    CodeLocals = 22,

    /// <seealso cref="UndertaleData.Strings"/>
    Strings = 23,

    /// <seealso cref="UndertaleData.EmbeddedImages"/>
    EmbeddedImages = 24,

    /// <seealso cref="UndertaleData.EmbeddedTextures"/>
    EmbeddedTextures = 25,

    /// <seealso cref="UndertaleData.TextureGroupInfo"/>
    TextureGroupInfo = 26,

    /// <seealso cref="UndertaleData.EmbeddedAudio"/>
    EmbeddedAudio = 27,

    /// <seealso cref="UndertaleData.Tags"/>
    Tags = 28,

    /// <seealso cref="UndertaleData.AnimationCurves"/>
    AnimationCurves = 29,

    /// <seealso cref="UndertaleData.Sequences"/>
    Sequences = 30,

    /// <seealso cref="UndertaleData.FeatureFlags"/>
    FeatureFlags = 31,

    /// <seealso cref="UndertaleData.FilterEffects"/>
    FilterEffects = 32,

    /// <seealso cref="UndertaleData.ParticleSystems"/>
    ParticleSystems = 33,

    /// <seealso cref="UndertaleData.ParticleSystemEmitters"/>
    ParticleSystemEmitters = 34,
}
